### PR TITLE
Fixes #15719 - check installation success before upgrades

### PR DESCRIPTION
**Pull request by iNecas**

### CODE REVIEW

File: hooks/post/30-upgrade.rb [hunk 1]
```diff
@@ -79,24 +79,28 @@ def fail_and_exit(message)
...
+      upgrade_step :update_puppet_repository_distributors, :long_running => true
+      upgrade_step :update_subscription_facet_backend_data, :long_running => true
+      upgrade_step :remove_gutterball
+    end
+
+    if [0,2].include? @kafo.exit_code
+      Kafo::Helpers.log_and_say :info, 'Katello upgrade completed!'
+    end
+  else
+    Kafo::Helpers.log_and_say :error, 'Katello upgrade failed at the installation phase, fix the error and re-run the upgrade.'
```
Comment (ehelms) [inline hooks/post/30-upgrade.rb]: `during the installation phase` sounds a bit more natural to me, and we could also leave off the 'Katello' aspect and just have it say 'Upgrade failed during the...`

File: hooks/post/30-upgrade.rb [hunk 2]
```diff
@@ -79,24 +79,28 @@ def fail_and_exit(message)
...
+      upgrade_step :elasticsearch_message
+      upgrade_step :add_export_distributor, :long_running => true
+      upgrade_step :remove_docker_v1_content, :long_running => true
+      upgrade_step :update_puppet_repository_distributors, :long_running => true
+      upgrade_step :update_subscription_facet_backend_data, :long_running => true
+      upgrade_step :remove_gutterball
+    end
+
+    if [0,2].include? @kafo.exit_code
+      Kafo::Helpers.log_and_say :info, 'Katello upgrade completed!'
```
Comment (ehelms) [inline hooks/post/30-upgrade.rb]: We could just change this to `Upgrade complete` or if you want to bake in some sound files: https://www.youtube.com/watch?v=cAez7KLXlfQ

File: hooks/post/30-upgrade.rb (hunk 1 above)
Comment (iNecas) [inline hooks/post/30-upgrade.rb]: Reworded

File: hooks/post/30-upgrade.rb (hunk 2 above)
Comment (iNecas) [inline hooks/post/30-upgrade.rb]: We need convert this into ASCIIart first, Let's do it as part of another issue. I'm ok with textual only version right now :) Reworded

**Comment by ehelms** [comment]
ACK - thanks @iNecas !

**Pull request merged**